pages/etherpad/actions/CreateAuthoredPad.js

```diff
@@ -1,40 +1,20 @@
 import { useState } from 'react';
-import getConfig from 'next/config';
-import { useRouter } from 'next/router';
 import { useTranslation } from 'react-i18next';
 
 import Form from '../../../components/UI/Form';
 import LoadingSpinnerInline from '../../../components/UI/LoadingSpinnerInline';
-import { useAuth } from '../../../lib/Auth';
-import { path as etherpadPath } from '../../../lib/Etherpad';
 
-export default function CreateAuthoredPad({ callbackDone, createWriteRoom }) {
-    const router = useRouter();
+export default function CreateAuthoredPad({ createPadAndOpen }) {
     const { t } = useTranslation('etherpad');
-
-    const auth = useAuth();
-    const etherpad = auth.getAuthenticationProvider('etherpad');
-
     const [padName, setPadName] = useState('');
     const [isLoading, setIsLoading] = useState(false);
 
     const createAuthoredPad = async () => {
         setIsLoading(true);
-        const padId = await etherpad.createPad(padName, 'public');
-
-        if (!padId) {
-            setIsLoading(false);
-
-            return;
-        }
-
-        const link = getConfig().publicRuntimeConfig.authProviders.etherpad.baseUrl + '/' + padId;
-        const roomId = await createWriteRoom(link, padName);
-        router.push(`${etherpadPath}/${roomId}`);
-
-        callbackDone && callbackDone();
```

Comment:
So these components used to receive a `callbackDone` function, which would be responsible for... closing the ServiceSubmenu after the process completed.
Without having tested this code: Is this still working? If so, can you explain me how?

(this is a rhetorical question of course; I'm pretty sure this is broken now and the service submenu remains open after you create a new password-protected / authored pad.......)

Reply:
is it possible that this has been broken for a while? `callbackDone` was never parsed to the component since it's been changed f178891b49deef15aa9ce6c8b8efcc764af4ca3a . Don't know how it would've closed before 🤔

Reply:
ah nvm, just found it 
https://github.com/medienhaus/medienhaus-spaces/blob/61126628f14c2ab1ebedf141146a3143e97c5071/components/UI/ServiceSubmenu.js#L49-L51